mmv1/templates/terraform/examples/cloud_run_service_tasks.tf.erb

```diff
@@ -0,0 +1,49 @@
+# [START cloud_run_service_tasks_service]
+resource "<%= ctx[:vars]['primary_resource_type'] %>" "<%= ctx[:primary_resource_id] %>" {
+    name     = "<%= ctx[:vars]['cloud_run_service_name'] %>"
+    location = "<%= ctx[:vars]['region'] %>"
+    project = "<%= ctx[:vars]['project_id'] %>"
+    template {
+      spec {
+            containers {
+                image = "gcr.io/cloudrun/hello"
+            }
+      }
+    }
+    traffic {
+      percent         = 100
+      latest_revision = true
+    }
+}
+# [END cloud_run_service_tasks_service]
+
+# [START cloud_run_service_tasks_sa]
+resource "google_service_account" "sa" {
+  account_id   = "cloud-run-task-invoker"
+  display_name = "Cloud Run Task Invoker"
+}
+# [END cloud_run_service_tasks_sa]
+
+# [START cloud_run_service_tasks_run_invoke_permissions]
+resource "google_cloud_run_service_iam_binding" "binding" {
+  location = google_cloud_run_service.default.location
+  service = google_cloud_run_service.default.name
+  role = "roles/run.invoker"
+  members = ["serviceAccount:${google_service_account.sa.email}"]
+}
+# [END cloud_run_service_tasks_run_invoke_permissions]
+
+# [START cloud_run_service_tasks_token_permissions]
+resource "google_project_iam_binding" "project" {
+  project = "<%= ctx[:vars]['project_id'] %>"
+  role    = "roles/iam.serviceAccountTokenCreator"
+  members = ["serviceAccount:${google_service_account.sa.email}"]
+}
+# [END cloud_run_service_tasks_token_permissions]
+
+# [START cloud_run_service_tasks_queue]
+resource "google_cloud_tasks_queue" "default" {
```

Comment:
nit: is this necessary for the example? it doesn't seem to be referenced anywhere or reference any of the other resources, and i don't see it in the linked example. i'm not super familiar with what's trying to be accomplished here, though, so i may be wrong. thanks!

Reply:
Apologies for not responding to this comment timely! So the idea for this was to present a step by step example for this tutorial https://cloud.google.com/run/docs/triggering/using-tasks#service-account-invoker

Alongside gcloud commands, this would be in its own Tf tab.